bpf/lib/nodeport.h

```diff
@@ -443,7 +443,7 @@ static __always_inline int dsr_reply_icmp6(struct __ctx_buff *ctx,
 	const __u32 l3_max = sizeof(*ip6) + orig_dgram;
 	__be16 type = bpf_htons(ETH_P_IPV6);
 	__s32 len_new = off + sizeof(*ip6) + orig_dgram;
-	__s32 len_old = ctx_full_len(ctx);
+	__u64 len_old = ctx_full_len(ctx);
 	void *data_end = ctx_data_end(ctx);
```

Comment:
nit: given we later on do `if (ctx_adjust_troom(ctx, -(len_old - len_new)) < 0)`, we should also convert len_new.